### PR TITLE
[tvOS][Top Shelf] Fix missing thumbs + clean up

### LINKED ISSUE #18409: tvOS movie posters not being populated
### Describe the bug
Movies are no longer showing images on the topshelf 

## Expected Behavior
Movie posters for recently added movies

## Actual Behavior
The posters are blank for recently added movies. TV shows are unaffected.

## Possible Fix
Changes made to the Topshelf handler might have introduced this bug.

### To Reproduce
Add new movies.
Go to the main menu 

### Debuglog
The debuglog can be found here: https://paste.kodi.tv/safexixugu.kodi

### Screenshots 
https://imgur.com/a/z5UZSq9

### Your Environment
Used Operating system: tvOS

 - [ ] Android
 - [ ] iOS
 - [x] tvOS
 - [ ] Linux
 - [ ] OSX
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: tvOS 13.4.8
 - Kodi version: 20200910-afbd3f9c-master-tvos

### PR DESCRIPTION
**Pull request by sy6sy2**
## Description

Since https://github.com/xbmc/xbmc/pull/18044/ (I guess), TopShelf thumbs are broken when user have more than one category (e.g. Recent movies and Recent Tv shows).
This PR fix issue https://github.com/xbmc/xbmc/issues/18409

## Motivation and Context

This bug is my fault because I forgot to adapt the piece of code responsible to delete old thumbs file. Currently, when we fill the TopShelf with Recent TV shows we delete all Recent movies thumbs....

My fix is to separate thumbs of each category in subdirectories. So when we update a TopShelf category we only remove old thumbs of this category.

I also added some comments and apply clean up just to understand what I did in the code later and also for others.

## How Has This Been Tested?

On my Apple TV 4K in sandboxed and no sandboxed mode.

## Types of change

- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/platform/darwin/tvos/TVOSTopShelf.mm [hunk 1]
```diff
@@ -62,14 +62,10 @@
...
-        [NSMutableSet setWithArray:[fileManager contentsOfDirectoryAtPath:storeUrl.path error:nil]];
-    std::string raPath = storeUrl.path.UTF8String;
 
-    // Shared dicts  (if we are sandboxed we use sharedDefaults, else we use sharedJailbreak)
-    auto sharedDefaults =
+    // Shared dicts (if we are sandboxed we use sharedSandbox, else we use sharedJailbreak)
+    auto sharedSandbox =
         isSandboxed ? [[NSUserDefaults alloc] initWithSuiteName:[tvosShared getSharedID]] : nil;
     const auto sharedJailbreakUrl = [storeUrl URLByAppendingPathComponent:@"shared.dict"];
     NSMutableDictionary* sharedJailbreak = nil;
```
Comment (fuzzard) [inline xbmc/platform/darwin/tvos/TVOSTopShelf.mm:71]: Is there really a need to have a sharedSandbox and a sharedJailbreak? They are both just dicts that are added/removed, with from what i can see the only difference being what happens in the end. Why not just use the one dict, and do a isSandboxed check for the synchronize/writeurl at the end?

Comment (sy6sy2) [reply xbmc/platform/darwin/tvos/TVOSTopShelf.mm:71]: `sharedSandbox` is a `NSUserDefaults` and `sharedJailbreak` is a `NSMutableDictionary`.
If I want to work with only one variable I need to:

1. If sandboxed --> Copy existing NSUserDefaults values in a NSMutableDictionary var. Else if jailbroken --> Open a NSMutableDictionary var from the existing `shared.dict` file
2. Update the NSMutableDictionary var according to the TopShelf category changes
3. If sandboxed --> clear all existing NSUserDefaults values and copy every key/value from the NSMutableDictionary var to the NSUserDefaults, then synchronize. Else if jailbroken --> Write the NSMutableDictionary var in `shared.dict` file.

Do I have good? If yes, in sandboxed case, the "ping pong" between NSUserDefaults and NSMutableDictionary seems heavy? No?

Comment (sy6sy2) [reply xbmc/platform/darwin/tvos/TVOSTopShelf.mm:71]: After some test I can confirm that `[[NSUserDefaults alloc] initWithSuiteName:[tvosShared getSharedID]]` doesn't work when Kodi is fake-signed. We need to keep two separate dict :-/

**Comment by sy6sy2** [comment]
Last push commit tested on both sandboxed and no-sandoboxed mode.
I tired to iIsolate TopShelf stuff inside NSUserDefaults with a special key (then in ServiceProvider we do not need to check for each key of NSUserDefaults).

If you have time to check my code. Thank you!

**Review by fuzzard — APPROVED** [review]

**Comment by sy6sy2** [comment]
I am thinking of doing the merge towards the end of the week if nobody is against.

**Pull request merged**